File: src/main/java/com/cf/service/LeaveworkService.java
package com.cf.service;
import java.util.List;

import com.cf.pojo.TbLeavework;
import com.cf.pojo.result.PageResult;
/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface LeaveworkService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbLeavework> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(TbLeavework leavework);
	
	
	/**
	 * 修改
	 */
	public void update(TbLeavework leavework);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbLeavework findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long [] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(TbLeavework leavework, int pageNum,int pageSize);
	
	/**
	 * 根据传入的id修改员工状态
	 * @param ids 传入id数组
	 * @param status 目标状态
	 */
	public void updateStatus(Long[] ids, String status);
	
	/**
	 *根据状态查找 
	 * @param status
	 * @return
	 */
	public List<TbLeavework> findByStatus(String status);
}

File: src/main/webapp/js/service/leaveworkService.js
//服务层
app.service('leaveworkService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../leavework/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../leavework/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../leavework/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../leavework/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../leavework/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../leavework/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../leavework/search.do?page='+page+"&rows="+rows, searchEntity);
	}    
	
	//状态更新
	this.updateStatus=function(ids,status){
		return $http.get('../leavework/updateStatus.do?ids='+ids+'&status='+status);
	}
	
	//状态更新
	this.findByStatus=function(status){
		return $http.get('../leavework/findByStatus.do?status='+status);
	}
});
